### PR TITLE
Otp verification for Business -18-11-2019

### PR DESCRIPTION
**Pull request by SMCIND**

### CODE REVIEW

File: src/app/register/business-register/business-register.component.html [hunk 1]
```diff
@@ -384,7 +491,141 @@ <h5 class="card-title text-center font-weight-bold">SIGN UP BUSINESS ACCOUNT</h5
...
+                        <option [ngValue]="+90">Turkey +90</option>
+                        <option [ngValue]="+971">United Arab Emirates +971</option>
+                        <option [ngValue]="+263">Zimbabwe +263</option>
+                        </select>        
+                </div> 
+                <div class="col-sm-4">
+                        <input type="text" minlength="10" maxlength="10" class="form-control" #v_phone_no required [(ngModel)]="model.primaryPhoneNumber" name="primaryPhoneNumber" #primaryPhoneNumber="ngModel">
+                </div>
+                <div class="col-sm-4">
+                        <button class="btn btn-small" type="button" (click)="sendVCode(ext_no.value,v_phone_no.value);" data-toggle="modal" data-target="#verificationstep1">{{send_resend}}</button>
```
Comment (saigoud1000) [inline src/app/register/business-register/business-register.component.html]: SEND button is not looking good. Can you make the SEND button same as sign up button.

Comment (saigoud1000) [reply src/app/register/business-register/business-register.component.html]: With background colour

File: src/app/register/business-register/business-register.component.html [hunk 2]
```diff
@@ -384,7 +491,141 @@ <h5 class="card-title text-center font-weight-bold">SIGN UP BUSINESS ACCOUNT</h5
...
+                <div class="col-sm-8 text-center">
+                        <label><h6>Verification Code will be send to the Phone Number below:</h6></label>  
+                </div>
+                
+        </div>
+        <div class="row" *ngIf="businessForm.form.valid">   
+                <div class="col-sm-1"></div>
+                <div class=" col-sm-3 form-group">
+                        <select type="text" class="form-control" #ext_no [(ngModel)]="model.primaryExtension" name="primaryExtension" #primaryExtension="ngModel" required>
+                        <option [ngValue]="null" disabled> --Select-- </option>
```
Comment (saigoud1000) [inline src/app/register/business-register/business-register.component.html]: Add **USA +1** to the extension.

File: src/app/register/business-register/business-register.component.html [hunk 3]
```diff
@@ -384,7 +491,141 @@ <h5 class="card-title text-center font-weight-bold">SIGN UP BUSINESS ACCOUNT</h5
                 </div>   
                 </div>     
         </div> 
+        <hr class="my-4" *ngIf="!businessForm.form.valid">
+        <div class="row mx-auto" *ngIf="businessForm.form.valid">
```
Comment (saigoud1000) [inline src/app/register/business-register/business-register.component.html]: Alignment is missing.

**Comment by saigoud1000** [comment]
Screen Shot 2019-11-17 at 7 40 30 PM

File: src/app/service/http-client.service.ts [hunk 4]
```diff
@@ -142,23 +145,15 @@ ChangeBusinessPassword(data): Observable<Business> {
   )
 } 
 
-// SMSVerification(data): Observable<any> {
```
Comment (saigoud1000) [inline src/app/service/http-client.service.ts]: Do we need this commented out code?

File: src/app/service/http-client.service.ts [hunk 5]
```diff
@@ -142,23 +145,15 @@ ChangeBusinessPassword(data): Observable<Business> {
...
-//   )
-// }  
-// GetVerificationCode(data): Observable<any> {
-//   let data_code = {"otp": data.otp};
-//   return this.httpClient.put<any>('/api/phonenumber/'+data.phone_no+'/otp/', JSON.stringify(data_code), this.httpOptions)
-//   .pipe(
-//     retry(1),
-//     catchError(this.errorHandl)
-//   )
+SMSVerification(data) {
```
Comment (saigoud1000) [inline src/app/service/http-client.service.ts]: Do the same for Customer Sign Up

**Comment by saigoud1000** [comment]
Screen Shot 2019-11-18 at 6 19 57 PM


I'm getting blank message after clicking on OTP SEND button.
**Note:**
API code is not deployed (failing)

File: src/app/register/customer-register/customer-register.component.html [hunk 6]
```diff
@@ -116,8 +126,137 @@ <h5 class="card-title text-center font-weight-bold">SIGN UP CUSTOMER ACCOUNT</h5
...
+                                </div>
+                                <br><br>
+                        </div>
+                <hr class="my-4" *ngIf="customerForm.form.valid">
+                
+                <div class="row"  *ngIf="v_step2_msg==''">
+                        <br>
+                        <button class="btn btn-lg btn-primary btn-block text-uppercase mx-auto" type="submit">Sign Up</button>
+                </div> 
+                <div class="row"  *ngIf="customerForm.form.valid && v_step2_msg=='OTP verified successfully and click on sign-up button to complete your registration.'">
```
Comment (saigoud1000) [inline src/app/register/customer-register/customer-register.component.html]: I guess you forgot add message in case of api error

**Comment by saigoud1000** [comment]
In case of unsuccessful API response(like 500Internal Server Error). We will ask user to check the phone number and click on send again

**Comment by SMCIND** [comment]
> In case of unsuccessful API response(like 500Internal Server Error). We will ask user to check the phone number and click on send again

Had handled such cases.. like error code 400 and error code 500

File: src/app/register/customer-register/customer-register.component.html (hunk 6 above)
Comment (SMCIND) [reply src/app/register/customer-register/customer-register.component.html]: It comes handled from typescript only, only success condition is handled in html as we need to submit the final form after successful otp verfication. Hence click action of the two buttons are different.

**Comment by SMCIND** [comment]
> Screen Shot 2019-11-18 at 6 19 57 PM
> 
> I'm getting blank message after clicking on OTP SEND button.
> **Note:**
> API code is not deployed (failing)

As messages are displayed as per response from backend and at times due to less internet speed the the modal gets displayed first and as message inside modal is coming from backend...it get delayed.

**Pull request merged**